test/recovery/recovery.c

```diff
@@ -0,0 +1,235 @@
+	 * space at the end to emphasize zero data.
+	 */
+	for (i = 0; i < MAX_VAL/4; i++)
+		buf[i] = (uint8_t)__wt_random(&rnd);
+
+	/*
+	 * Run in the home directory so that the records file is in there too.
+	 */
+	chdir(home);
+	if ((ret = wiredtiger_open(NULL, NULL, ENV_CONFIG, &conn)) != 0)
+		testutil_die(ret, "wiredtiger_open");
+	if ((ret = conn->open_session(conn, NULL, NULL, &session)) != 0)
+		testutil_die(ret, "WT_CONNECTION:open_session");
+	if ((ret = session->create(session,
+	    uri, "key_format=Q,value_format=u")) != 0)
+		testutil_die(ret, "WT_SESSION.create: %s", uri);
+	if ((ret =
+	    session->open_cursor(session, uri, NULL, NULL, &cursor)) != 0)
+		testutil_die(ret, "WT_SESSION.open_cursor: %s", uri);
+
+	/*
+	 * Keep a separate file with the records we wrote for checking.
+	 */
+	unlink(RECORDS_FILE);
+
+	if ((fd = open(RECORDS_FILE, O_CREAT | O_WRONLY, 0777)) == -1)
+		testutil_die(errno, "open");
+
+	/*
+	 * Write data into the table until we are killed by the parent.
+	 * The data in the buffer is already set to random content.
+	 */
+	data.data = buf;
+	for (i = 0;; ++i) {
+		data.size = __wt_random(&rnd) % MAX_VAL;
+		cursor->set_key(cursor, i);
+		cursor->set_value(cursor, &data);
+		if ((ret = cursor->insert(cursor)) != 0)
+			testutil_die(ret, "WT_CURSOR.insert");
+		if (write(fd, &i, sizeof(i)) != sizeof(i))
+			testutil_die(errno, "write");
+	}
+}
+
+extern int __wt_optind;
+extern char *__wt_optarg;
+
+int
+main(int argc, char *argv[])
+{
+	WT_CONNECTION *conn;
+	WT_CURSOR *cursor;
+	WT_SESSION *session;
+	WT_RAND_STATE rnd;
+	uint64_t key;
+	uint32_t absent, count;
+	size_t rd;
+	int ch, fd, status, ret, timeout;
+	pid_t pid;
+	char *working_dir;
+
+	if ((progname = strrchr(argv[0], DIR_DELIM)) == NULL)
+		progname = argv[0];
+	else
+		++progname;
+
+	working_dir = NULL;
+	timeout = 10;
+	while ((ch = __wt_getopt(progname, argc, argv, "h:t:")) != EOF)
+		switch (ch) {
+		case 'h':
+			working_dir = __wt_optarg;
+			break;
+		case 't':
+			timeout = atoi(__wt_optarg);
+			break;
+		default:
+			usage();
+		}
+	argc -= __wt_optind;
+	argv += __wt_optind;
+	if (argc != 0)
+		usage();
+
+	testutil_work_dir_from_path(home, 512, working_dir);
+	testutil_make_work_dir(home);
+
+	/*
+	 * Fork a child to insert as many items.  We will then randomly
+	 * kill the child, run recovery and make sure all items we wrote
+	 * exist after recovery runs.
+	 */
+	if ((pid = fork()) < 0)
+		testutil_die(errno, "fork");
+
+	if (pid == 0) { /* child */
+		fill_db();
+		return (EXIT_SUCCESS);
+	}
+
+	/* parent */
+	__wt_random_init(&rnd);
+	/* Sleep for the configured amount of time before killing the child. */
+	printf("Parent: sleep %" PRIu32 " seconds, then kill child\n", timeout);
+	sleep(timeout);
+
+	/*
+	 * !!! It should be plenty long enough to make sure more than one
+	 * log file exists.  If wanted, that check would be added here.
+	 */
+	kill(pid, SIGKILL);
```

Comment:
On my machine it takes over 10 seconds for this kill command to complete (above the sleep timeout). Do you think we could speed that up? Maybe throttling the worker would get the signal to complete faster.

Also - using sub processes and signals means that this test is going to be difficult to port to Windows. I don't have a good alternative - since we want to force an unclean shutdown.